tests/sentry/services/test_hybrid_cloud.py

```diff
@@ -0,0 +1,75 @@
+from collections import deque
+
+from sentry.services.hybrid_cloud import (
+    SiloDataInterface,
+    UnsetType,
+    app,
+    auth,
+    identity,
+    integration,
+    log,
+    notifications,
+    organization,
+    organization_mapping,
+    tombstone,
+    user,
+    user_option,
+)
+from sentry.testutils import TestCase
+
+
+class SiloDataInterfaceTest(TestCase):
+    INTERFACE_CLASSES = frozenset(
+        [
+            app.ApiSentryApp,
+            app.ApiSentryAppInstallation,
+            auth.ApiAuthIdentity,
+            auth.ApiAuthProvider,
+            auth.ApiAuthProviderFlags,
+            auth.ApiAuthState,
+            auth.ApiMemberSsoState,
+            auth.ApiOrganizationAuthConfig,
+            identity.APIIdentity,
+            identity.APIIdentityProvider,
+            integration.APIIntegration,
+            integration.APIOrganizationIntegration,
+            log.AuditLogEvent,
+            log.UserIpEvent,
+            notifications.ApiNotificationSetting,
+            organization.ApiOrganization,
+            organization.ApiOrganizationFlags,
+            organization.ApiOrganizationMember,
+            organization.ApiOrganizationMemberFlags,
+            organization.ApiOrganizationSummary,
+            organization.ApiProject,
+            organization.ApiTeam,
+            organization.ApiTeamMember,
+            organization.ApiUserOrganizationContext,
+            organization_mapping.APIOrganizationMapping,
+            organization_mapping.ApiOrganizationMappingUpdate,
+            tombstone.ApiTombstone,
+            user.APIAvatar,
+            user.APIUser,
+            user.APIUserEmail,
+            user_option.ApiUserOption,
+        ]
+    )
+
+    def test_schema_generation(self):
+        for api_type in self.INTERFACE_CLASSES:
+            # We're mostly interested in whether an error occurs
+            schema = api_type.schema_json()
+            assert schema
+
+    def test_interface_class_coverage(self):
+        subclasses = set()
+        stack = deque([SiloDataInterface])
+        while stack:
+            next_class = stack.pop()
+            if next_class not in subclasses:
+                subclasses.add(next_class)
+                stack += next_class.__subclasses__()
+
+        subclasses.difference_update({SiloDataInterface, UnsetType})
+        uncovered = subclasses.difference(self.INTERFACE_CLASSES)
+        assert uncovered == set(), "SiloDataInterface subclasses exist that are not tested"
```

Comment:
I wasn't sure if I wanted to gather up all `SiloDataInterface` subclasses or list them explicitly, so I kind of tried both at once. We could choose one approach or the other if this seems too heavy-handed.

Reply:
Harvesting the classes using reflection/inference should be ok. We could also have assertions during RPC serialization that entities extend `SiloDataInterface` as well.

Having to maintain an explicit list could get tedious over time.

Reply:
One approach we can consider in the future -- since we will already have a registery of every service, we could just use that registery to do the top level open api schema generation in a loop here?